packages/core/src/shared/sam/localLambdaRunner.ts

```diff
@@ -290,15 +287,16 @@ export async function runLambdaFunction(
         getLogger().info(localize('AWS.output.sam.local.startRun', 'Preparing to run locally: {0}', config.handlerName))
     }
 
-    const envVars = {
+    const envVars = await getSpawnEnv({
+        ...process.env,
         ...(config.aws?.region ? { AWS_DEFAULT_REGION: config.aws.region } : {}),
-    }
+    })
 
     const settings = SamCliSettings.instance
     const timer = new Timeout(settings.getLocalInvokeTimeout())
 
     // TODO: refactor things to not mutate the config
-    config.templatePath = await buildLambdaHandler(timer, await getSpawnEnv(envVars), config, settings)
```

Comment:
1. Were you able to reproduce the issue locally in your window instance.
2. I think this is where the issue is from but this file remains constant across versions. I think we need to figure out what is changing the executable path before it get here for windows.

Reply:
1: Yes I can reproduce.
2: see comment above